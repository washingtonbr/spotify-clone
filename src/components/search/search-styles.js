import styled from 'styled-components';

export const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  border-bottom: 2px solid ${(props) => props.theme.secondaryColor};
  padding: 0 8px;
  margin-bottom: 32px;

  @media(min-width: 768px) {
    margin-bottom: 62px;
  }

  @media (min-width: 992px) {
    margin-top: 70px;
    margin-bottom: 82px;
  }
`;

export const Label = styled.label`
  font-size: 12px;
  margin-bottom: 10px;

  @media(min-width: 768px) {
    font-size: 14px;
  }

  @media(min-width: 992px) {
    font-size: 16px;
  }
`;

export const Input = styled.input`
  height: 28px;
  font-size: 24px;
  font-weight: 700;
  line-height: 28px;
  color: ${(props) => props.theme.secondaryColor};
  background-color: transparent;
  border: 0;
  margin-bottom: 14px;

  @media(min-width: 768px) {
    height: 33px;
    font-size: 28px;
    line-height: 33px;
  }

  @media(min-width: 992px) {
    height: 56px;
    font-size: 48px;
    line-height: 56px;
  }
`;
