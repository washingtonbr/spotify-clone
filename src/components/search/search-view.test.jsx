import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import SearchView from './search-view';
import { Input } from './search-styles';

const mockDoSearch = jest.fn();

describe('SearchView', () => {
  test('renders correctly', () => {
    const tree = renderer
      .create(
        <SearchView
          searchText="Eminem"
          doSearch={mockDoSearch}
        />,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('should call search function', () => {
    const wrapper = shallow(
      <SearchView
        searchText="Eminem"
        doSearch={mockDoSearch}
      />,
    );

    wrapper
      .find(Input)
      .first()
      .simulate('change', {
        target: {
          value: 'Eminem',
        },
      });

    expect(mockDoSearch).toBeCalledTimes(1);
  });
});
