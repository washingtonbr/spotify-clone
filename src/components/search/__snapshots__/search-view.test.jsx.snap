// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchView renders correctly 1`] = `
<div
  className="sc-AxjAm CMpkT"
>
  <label
    className="sc-AxirZ dwykhk"
    htmlFor="search-field"
  >
    Busque por artistas, álbuns ou músicas
  </label>
  <input
    className="sc-AxiKw kUCIYV"
    id="search-field"
    onChange={[Function]}
    placeholder="Comece a escrever..."
    value="Eminem"
  />
</div>
`;
